source/isaaclab/isaaclab/assets/utils/wrench_composer.py

```diff
@@ -0,0 +1,280 @@
+        """Wrench composer.
+
+        This class is used to compose forces and torques at the body's center of mass frame.
+        It can compose global wrenches and local wrenches. The result is always in the center of mass frame of the body.
+
+        Args:
+            num_envs: Number of environments.
+            num_bodies: Number of bodies.
+            device: Device to use.
+            asset: Asset to use. Defaults to None.
+        """
+        self.num_envs = num_envs
+        self.num_bodies = num_bodies
+        self.device = device
+        self._asset = asset
+        self._active = False
+
+        if (self._asset.__class__.__name__ == "Articulation") or (self._asset.__class__.__name__ == "RigidObject"):
+            self._get_com_fn = lambda a=self._asset: a.data.body_com_pos_w[..., :3]
+        elif self._asset.__class__.__name__ == "RigidObjectCollection":
+            self._get_com_fn = lambda a=self._asset: a.data.object_com_pos_w[..., :3]
+        elif self._asset is None:
+            self._get_com_fn = lambda: wp.zeros((self.num_envs, self.num_bodies), dtype=wp.vec3f, device=self.device)
+        else:
+            raise ValueError(f"Unsupported asset type: {self._asset.__class__.__name__}")
+
+        self._com_positions_updated = False
+
+        self._composed_force_b = wp.zeros((num_envs, num_bodies), dtype=wp.vec3f, device=device)
+        self._composed_torque_b = wp.zeros((num_envs, num_bodies), dtype=wp.vec3f, device=device)
+        self._com_positions = wp.zeros((num_envs, num_bodies), dtype=wp.vec3f, device=device)
+        # Pinning the composed force and torque to the torch tensor to avoid copying the data to the torch tensor every time.
+        self._composed_force_b_torch = wp.to_torch(self._composed_force_b)
+        self._composed_torque_b_torch = wp.to_torch(self._composed_torque_b)
+
+    @property
+    def active(self) -> bool:
+        """Whether the wrench composer is active."""
+        return self._active
+
+    @property
+    def composed_force(self) -> wp.array:
+        """Composed force at the body's center of mass frame.
+
+        .. note:: If some of the forces are applied in the global frame, the composed force will be in the center
+        mass frame of the body.
+
+        Returns:
+            wp.array: Composed force at the body's center of mass frame. (num_envs, num_bodies, 3)
+        """
+        return self._composed_force_b
+
+    @property
+    def composed_torque(self) -> wp.array:
+        """Composed torque at the body's center of mass frame.
+
+        .. note:: If some of the torques are applied in the global frame, the composed torque will be in the center
+        mass frame of the body.
+
+        Returns:
+            wp.array: Composed torque at the body's center of mass frame. (num_envs, num_bodies, 3)
+        """
+        return self._composed_torque_b
+
+    @property
+    def composed_force_as_numpy(self) -> np.ndarray:
```

Comment:
Why add these numpy properties? We don't plan to support that, and if users need it, they can do `.numpy` themselves.